[ChenLeiyu] iP

ChenLeiyu opened the pull request:
# [Froggy](https://github.com/nus-cs2103-AY2425S1/ip/pull/427)

A bot that helps you keep track of tasks:
- text-based
- fast input
- lightweight

Here are some features:
1. Add tasks of up to **3** types!
2. Mark tasks as ~~incomplete~~ complete! 😇 
3. Saves task progress _through sessions_!

Just do the following steps:
- [ ] Download program
- [ ] Run program

Just run the `main` method:
```ruby
public class Froggy {
    public static void main(String[] args) {
        Froggy.run("./data/taskList.txt");
    }
}
```

> [!NOTE]
> The author is not responsible for any missed deadline or event incurred from using this program.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ChenLeiyu committed:
Change chatbot name from Duke to Froggy
Add basic greeting and exit messages

ChenLeiyu committed:
Add basic chatbot input

- Add input
- Add chatbot termination on "bye"

ChenLeiyu committed:
Add echo function of chatbot

ChenLeiyu committed:
Add task input and listing

- Change greeting text to explain chatbot commands
- Change text line formatting

ChenLeiyu committed:
Implement Task class

ChenLeiyu committed:
Add marking and unmarking tasks

ChenLeiyu committed:
Add toString function to display tasks

ChenLeiyu committed:
Add todo, event, deadline

ChenLeiyu committed:
Add text UI testing

ChenLeiyu committed:
Add some error handling

- Add error handling for non-integer after 'mark' and 'unmark'
- Add error message for 'todo ' with no description
- Add error message for unrecognised input

ChenLeiyu committed:
Add delete

ChenLeiyu committed:
Improve error handling and messages for mark, unmark and delete commands

- Add specific error message for e.g. 'mark' and 'mark '.
- Improve error message given

ChenLeiyu committed:
Fix if-else formatting

ChenLeiyu committed:
Implement saving tasks in hard disk

Add loadTasks and SaveTasks methods to load or save to hard drive.
Program saves to ./data/taskList.txt. loadTasks() called initially and
SaveTasks() called every input.

Add toTxt() method to Task and respective subclasses. This handles
how a task is stored in the .txt file on hard drive.

ChenLeiyu committed:
Add handling case where data file does not exist

ChenLeiyu committed:
Remove taskList.txt from tracking by git

ChenLeiyu committed:
Merge branch 'branch-level-7'

ChenLeiyu committed:
Add storing deadlines as time if format is right

If deadline description is in the form yyy-mm-dd, it will be stored as date and
printed as MMM dd yyy e.g.(input: "2019-10-15", printed:"Oct 15 2019")

ChenLeiyu committed:
Merge branch 'branch-Level-8'

ChenLeiyu committed:
Refactor code to extract out closely related code as classes

-Ui class deals with interactions
-Storage class deals with loading and saving from storage
-Parser class deals with user input
-TaskList handles operations relating to task list
-Command and its sub-classes for each command

ChenLeiyu committed:
Add own Exception class

ChenLeiyu committed:
Merge branch 'Branch-A-MoreOOP'

ChenLeiyu committed:
Rename directory from src/main/java to src/main/froggy
Update package declerations

ChenLeiyu committed:
Delete commented code

ChenLeiyu committed:
Remove taskList.txt file from tracking

ChenLeiyu committed:
Merge branch 'Branch-A-Packages'

ChenLeiyu committed:
Change package name to correct file directory

ChenLeiyu committed:
Add InvalidCommand class

ChenLeiyu committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ChenLeiyu committed:
Change file directory and packaging

Chang classes directory from src/main/froggy to
src/main/java/froggy and change package to froggy

ChenLeiyu committed:
Merge branch 'Branch-A-Gradle'

ChenLeiyu committed:
Add Deadline and Storage JUnit tests

ChenLeiyu committed:
Merge branch 'Branch-A-JUnit'

ChenLeiyu committed:
Change Jar name to Froggy in gradle build

ckaayy commented on src/main/java/froggy/AddEventCommand.java:
```diff
@@ -0,0 +1,31 @@
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        int index = input.toLowerCase().indexOf("/from ");
+        int index2 = input.toLowerCase().indexOf("/to ");
```
Should we avoid using numbers to distinguish variable names?

ckaayy commented on src/main/java/froggy/Storage.java:
```diff
@@ -0,0 +1,75 @@
+package froggy;
+import java.io.BufferedReader;
```
Should there be a line separator?

ckaayy commented on src/main/java/froggy/Storage.java:
```diff
@@ -0,0 +1,75 @@
+                            taskList.add(newDeadline);
+                            break;
+                        case 'E':
+                            int index1 = line.indexOf('|');
+                            int index2 = line.indexOf('|', index1 + 1);
```
Should we avoid using numbers to distinguish variables?

ckaayy commented on src/test/java/froggy/StorageTest.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @Test
+    public void saveTasks_todoTaskList_correctTxtFile() throws Exception {
```
I like how you followed the test method naming convention!

ckaayy reviewed (commented):
LGTM!:)
Just a few nits

ChenLeiyu committed:
Add find command

Add header comments for some classes

ChenLeiyu committed:
Merge branch 'branch-Level-9'

chongsun2002 commented on src/main/java/froggy/AddEventCommand.java:
```diff
@@ -0,0 +1,39 @@
+        int index = input.toLowerCase().indexOf("/from ");
+        int index2 = input.toLowerCase().indexOf("/to ");
+        if (index != -1) {
+            String desc = input.substring(6, index - 1);
+            String from = input.substring(index + 6, index2 - 1);
+            String to = input.substring(index2 + 4);
+            Event current = new Event(desc, from, to);
```
Are there better ways to name these strings and the event?

chongsun2002 commented on src/main/java/froggy/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class Deadline extends Task {
+
+    protected final String by;
+    protected LocalDate date;
```
Can these have different names that are more descriptive? I see these in other places in your code.

chongsun2002 replied:
```suggestion
    protected LocalDate dueByDateAsLocalDate;
```

chongsun2002 commented on src/main/java/froggy/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class Deadline extends Task {
+
+    protected final String by;
```
```suggestion
    protected final String dueByDateAsString;
```

chongsun2002 commented on src/main/java/froggy/Storage.java:
```diff
@@ -0,0 +1,75 @@
+package froggy;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
```
I like how your imports are sorted by their purposes!

chongsun2002 reviewed (commented):
Overall, your code is well abstracted and fulfills its purpose. Just a few style nits to fix. LGTM.

ChenLeiyu committed:
Add checkstyle.xml and suppressions.xml

ChenLeiyu committed:
Merge branch 'branch-A-CheckStyle'

ChenLeiyu committed:
Add GUI to chatbot

ChenLeiyu committed:
Merge branch 'branch-Level-10'

ChenLeiyu committed:
Add greeting when opening Froggy

ChenLeiyu committed:
Delete command: fix tasks not saving after delete

Delete command deletes from TaskList but does not update the local .txt file.
If the user deletes and exits without another command to save,
the deletion is not saved. This is inconsistent with other commands
that save the TaskList after successful execution.

Let's save TaskList to local file after every successful deletion.

ChenLeiyu committed:
Add task commands: improve input parsing

Add task commands required a space before and after additional identifiers
if not the character before will not be saved. This is not intuitive for the user.

Let's start parsing the description right before and after the identifiers such as "/by".

ChenLeiyu committed:
Refactor code

- Extract method to check if index is in range to TaskList.
- Remove unnecessary prints when executing an getting string in commands.
- Remove unnecessary comments.

ChenLeiyu committed:
Merge pull request #1 from ChenLeiyu/branch-A-CodeQuality

A-CodeQuality

ChenLeiyu committed:
Merge branch 'master' of https://github.com/ChenLeiyu/ip

ChenLeiyu committed:
Add prevention of adding of duplicate classes

ChenLeiyu committed:
Merge branch 'branch-BCD-Extension'

ChenLeiyu committed:
Fix bug when input is 'todo'

ChenLeiyu committed:
Remove dead code

ChenLeiyu committed:
Add error for invalid file format

ChenLeiyu committed:
Add program name in gui window title

ChenLeiyu committed:
Update README and use guide

ChenLeiyu committed:
Fix image not displayed on README

ChenLeiyu committed:
Add Javadoc comments to methods

ChenLeiyu committed:
Update user guide
